resources/scripts/ttx_diff.py

```diff
@@ -407,17 +475,30 @@ def main(argv):
 
     comparisons = (FLAGS.compare,)
     if comparisons == ("both",):
+        if FLAGS.json:
+            sys.exit("JSON output does not support multiple comparisons (try --compare default|gftools)")
         comparisons = (_COMPARE_DEFAULTS, _COMPARE_GFTOOLS)
 
     for compare in comparisons:
         build_dir = (root / "build" / compare).relative_to(root)
         build_dir.mkdir(parents=True, exist_ok=True)
-        print(f"Compare {compare} in {build_dir}")
+        maybe_print(f"Compare {compare} in {build_dir}")
+
+        try:
+            fontc_ttf = build_fontc(source.resolve(), build_dir, compare)
+        except Exception as e:
+            handle_compiler_error(e, "fontc", FLAGS.json)
+        try:
+            fontmake_ttf = build_fontmake(source.resolve(), build_dir, compare)
 
-        fontc_ttf = build_fontc(source.resolve(), build_dir, compare)
-        fontmake_ttf = build_fontmake(source.resolve(), build_dir, compare)
+        except Exception as e:
+            handle_compiler_error(e, "fontmake", FLAGS.json)
```

Comment:
The blocks are identical other than which build function is called, could be a function.